src/tiledb/cloud/vcf/split.py

```diff
@@ -0,0 +1,343 @@
+    """
+
+    import logging
+
+    import tiledb
+    from tiledb.cloud.dag import DAG
+    from tiledb.cloud.dag import Mode
+    from tiledb.cloud.rest_api import RetryStrategy
+    from tiledb.cloud.utilities import get_logger
+    from tiledb.cloud.utilities import run_dag
+    from tiledb.cloud.version import version
+
+    logger = get_logger(level=logging.DEBUG if verbose else logging.INFO)
+
+    logger.debug(
+        "tiledb.cloud=%s, tiledb=%s, libtiledb=%s",
+        version,
+        tiledb.version(),
+        tiledb.libtiledb.version(),
+    )
+
+    graph = DAG(
+        mode=Mode.BATCH,
+        namespace=namespace,
+        name="Split VCF",
+        max_workers=max_workers,
+        retry_strategy=RetryStrategy(
+            limit=retry_count,
+            retry_policy="Always",
+        ),
+    )
+
+    if not samples:
+        logger.debug("Isolating all samples from aggregate VCF.")
+
+        samples = graph.submit(
+            ls_samples,
+            vcf_uri=vcf_uri,
+            config=config,
+            access_credentials_name=acn,
+            name="List Samples",
+            resources={
+                "cpu": "1",
+                "memory": "5Gi",
+            },
+        )
+
+        graph.submit_udf_stage(
+            split_one_sample,
+            vcf_uri=vcf_uri,
+            sample=samples,
+            output_uri=output_uri,
+            config=config,
+            expand_node_output=samples,
+            access_credentials_name=acn,
+            name="Split Sample From VCF",
+            resources=resources,
+        )
+    else:
+        logger.debug(f"Isolating {len(samples)} samples from aggregate VCF.")
+
+        # isolate specified samples only
+        for s in samples:
+            graph.submit(
+                split_one_sample,
+                vcf_uri=vcf_uri,
+                sample=s,
+                output_uri=output_uri,
+                config=config,
+                access_credentials_name=acn,
+                name="Split Sample From VCF",
+                resources=resources,
+            )
+
+    if compute:
+        run_dag(graph, wait=False, debug=verbose)
+
+        logger.info(
+            "Task Graph submitted -"
+            " https://cloud.tiledb.com/activity/taskgraphs/%s/%s",
+            graph.namespace,
+            graph.server_graph_uuid,
+        )
+
+    logger.debug(f"Returning DAG: {graph.name}")
+
+    return graph
+
+
+if __name__ == "__main__":
+    import tiledb.cloud
+
+    tiledb.cloud.client.config.config.host = "https://us-west-2.aws.api.tiledb.com"
+
+    graph = split_vcf(
+        vcf_uri="s3://tiledb-spencer/projects/horses/raw/thesis_intersect_pub.decomposed.vcf.gz",
+        output_uri="s3://tiledb-spencer/projects/vcf-isolate-dev/isolated-singles",
+        config={"vfs.s3.region": "us-west-2"},
+        samples=["A1543"],
+        namespace="TileDB-Inc",
+        acn="tiledb-cloud-sandbox-role",
+        compute=True,
+    )
```

Comment:
Please remove the testing code. It could be added to the manually run pytests, see https://github.com/TileDB-Inc/TileDB-Cloud-Py/blob/main/tests/test_vcf.py.

Reply:
Yeah I will! This PR is still draft before ready for approval. i'll add unit tests as well